tls/extensions/s2n_psk_key_exchange_modes.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <sys/param.h>
+#include <stdint.h>
+
+#include "tls/s2n_tls_parameters.h"
+#include "tls/extensions/s2n_client_psk.h"
+#include "utils/s2n_safety.h"
+
+#define PSK_KEY_EXCHANGE_MODE_SIZE sizeof(uint8_t)
+
+static bool s2n_psk_key_exchange_modes_should_send(struct s2n_connection *conn);
+static int s2n_psk_key_exchange_modes_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_psk_key_exchange_modes_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
+
+const s2n_extension_type s2n_psk_key_exchange_modes_extension = {
+    .iana_value = TLS_EXTENSION_PSK_KEY_EXCHANGE_MODES,
+    .is_response = false,
+    .send = s2n_psk_key_exchange_modes_send,
+    .recv = s2n_psk_key_exchange_modes_recv,
+    .should_send = s2n_psk_key_exchange_modes_should_send,
+    .if_missing = s2n_extension_noop_if_missing,
+};
+
+static bool s2n_psk_key_exchange_modes_should_send(struct s2n_connection *conn)
+{
+    /* Only send a psk_key_exchange_modes extension if a psk extension is also being sent */
+    return s2n_client_psk_should_send(conn);
```

Comment:
We could add a  `  notnull_check(conn);` check here.

Reply:
The check would be unnecessary; conn isn't dereferenced here.